skyplane/api/dataplane.py

```diff
@@ -156,13 +156,34 @@ def provision(
             is_ibmcloud_used = any(n.region_tag.startswith("ibmcloud:") for n in self.topology.get_gateways())
 
             # create VMs from the topology
+            auths = {
+                "aws": compute.AWSAuthentication,
+                "azure": compute.AzureAuthentication,
+                "gcp": compute.GCPAuthentication,
+                "ibmcloud": compute.IBMCloudAuthentication,
+            }
             for node in self.topology.get_gateways():
                 cloud_provider, region = node.region_tag.split(":")
+                spot = getattr(self.transfer_config, f"{cloud_provider}_use_spot_instances")
+                vm_type = getattr(self.transfer_config, f"{cloud_provider}_instance_class")
+
+                # Fall back to smaller vm if necessary
+                auth = auths[cloud_provider]
+                quota_limit = auth.get_quota_limits_for(region, spot)
+                if quota_limit is not None:
+                    smaller_vm = auth.fall_back_to_smaller_vm_if_neccessary(instance_type=vm_type, quota_limit=quota_limit)
+                    if smaller_vm is not None:
+                        vm_type = smaller_vm
+                        logger.warning(f"Falling back to {smaller_vm} at node {region} due to the vCPU quota limit {quota_limit}")
```

Comment:
What do you think of the warning message? Do you think it makes sense to write something like this in addition to the existing message: 
`Go to https://skyplane.org/en/latest/increase_vcpus.html for more instructions on how to increase cloud vCPU limits`